doc/reference/std/errors.md

```diff
@@ -1,64 +1,421 @@
+
+### bad-argument-error?
+```scheme
+(def bad-argument-error? BadArgument?
+```
+
+same as `BadArgument?`
+
+### IOError
+```scheme
+(defclass (IOError StackTrace Error) ())
+```
+
+Error indicating an IO condition.
+
+### IOError?
+```scheme
+(IOError? exn)
+```
+
+Predicate testing whether the error is an IO condition.
+
+### io-error?
+```scheme
+(def io-error? IOError?
+```
+
+same as `IOError?`
+
+### PrematureEndOfInput
+```scheme
+(defclass (PrematureEndOfInput IOError) ())
+```
+
+Error indicating that input was exhausted prematurely; similar to EOF
+exceptions in other languages.
+
+### PrematureEndOfInput?
+```scheme
+(PrematureEndOfInput? obj)
+```
+
+Predicate testing whether the error is a premature end of input condition.
+
+### premature-end-of-input-error?
+```scheme
+(def permature-end-of-input? PrematureEndOfInput?)
+```
+
+Same as `PrematureEndOfInput?`.
+
+### IOClosed
+```scheme
+(deferror-class (IOClosed IOError))
+```
+
+Error indicating that the IO sink or source has been closed and thus
+no further IO operations can be performed.
+
+### IOClosed?
+```scheme
+(IOClosed? obj)
+```
+
+Predicate testing whether the error is an IO closed condition.
+
+### io-closed-error?
+```scheme
+(def io-closed-error? IOClosed?)
+```
+
+Same as `IOClosed?`.
+
+### Timeout
+```scheme
+(deferror-class Timeout)
+```
+
+Error indicating that some operation has timed out.
+
+### Timeout?
+```scheme
+(Timeout? obj)
+```
+
+Predicate testing whether the error is a timeout.
+
+### timeout-error?
+```scheme
+(def timeout-error? Timeout?)
+```
+
+Same as `Timeout?`.
+
+### ContextError
+```scheme
+(deferror-class ContextError)
+```
+
+Error indicating that some operation was performed out of context.
+
+### ContextError?
+```scheme
+(ContextError? obj)
+```
+
+Predicate testing whether the error is a context error
+
+### context-error?
+```scheme
+(def context-error? ContextError?)
+```
+
+Same as `ContextError?`.
+
+### KeyError
+```scheme
+(deferror-class KeyError)
+```
+
+Error indicating that some lookup operation failed because a key was unbound.
+
+### KeyError?
+```scheme
+(KeyError? obj)
+```
+
+Predicate testing whether the error is a key error
+
+### key-error?
+```scheme
+(def key-error? KeyError?)
+```
+
+Same as `KeyError?`.
+
+## Raising exceptions
+
+The library redefines `raise` and `error` so that it fills stack
+traces where appropriate.
+It also provides some utility procedure for raising particular errors
+without caring about the particulars of the condition system.
+
+### error
+```scheme
+(error message irritant ...)
+```
+
+Like basic Scheme's `error`, but raises with stack trace.
+
+### raise
+```scheme
+(raise exn)
+```
+
+Raises an exception condition.
+
+If the exception object is an instance of `StackTrace`, then unless
+there is already a continuation because this is a re-raise, the
+continuation is captured and stored in the `continuation` slot.
+
+### raise-bad-argument
```

Comment:
At least in this file, replace all the `what`'s by `message` ?

Reply:
sure, done in source already.